controllers/common/lock.go

```diff
@@ -0,0 +1,44 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package common
+
+import (
+	"sync"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// PoliciesLock allows locking at the root policy level so that multiple controllers/goroutines don't race with each
+// other and cause unnecessary reconcile retries.
+type PoliciesLock struct {
```

Comment:
I want to think more about this lock in general. I might need some time to stew with it, or I might need to ask questions directly (so be prepared ;) )